dependencies/src/main/java/com/google/cloud/tools/opensource/classpath/ClassDumper.java

```diff
@@ -142,6 +147,97 @@ static ClassDumper create(List<Path> jarFilePaths) {
     return methodReferences.build();
   }
 
+  static SymbolsInFile scanClassSymbolTable(JavaClass javaClass) {
+    SymbolsInFile.Builder symbolTableBuilder = SymbolsInFile.builder();
+    ImmutableSet.Builder<MethodSymbolReference> methodReferences =
+        symbolTableBuilder.methodReferencesBuilder();
+    ImmutableSet.Builder<FieldSymbolReference> fieldReferences =
+        symbolTableBuilder.fieldReferencesBuilder();
+
+    // TODO(suztomo): Read class references (inheritance) from javaClass file
+    ImmutableSet.Builder<ClassSymbolReference> classReferences =
+        symbolTableBuilder.classReferencesBuilder();
+
+    String sourceClassName = javaClass.getClassName();
+    ConstantPool constantPool = javaClass.getConstantPool();
+    Constant[] constants = constantPool.getConstantPool();
+    for (Constant constant : constants) {
+      if (constant == null) {
+        continue;
+      }
+      byte constantTag = constant.getTag();
+      switch (constantTag) {
+        case Const.CONSTANT_Methodref:
+          ConstantMethodref constantMethodref = (ConstantMethodref) constant;
+          methodReferences.add(constantToMethodReference(constantMethodref, constantPool,
+              sourceClassName));
+          break;
+        case Const.CONSTANT_Fieldref:
+          ConstantFieldref constantFieldref = (ConstantFieldref) constant;
+          fieldReferences.add(constantToFieldReference(constantFieldref, constantPool,
+              sourceClassName));
+          break;
+        case Const.CONSTANT_Class:
+          // TODO(suztomo): handle class reference
+          break;
+        default:
+          break;
+      }
+    }
+
+    Set<String> definedClassNames = Sets.newHashSet(listInnerClassNames(javaClass));
```

Comment:
Cab you use a construcctor here instead?

Reply:
If the `SymbolTable` builder gave you an `ImmutableSet.Builder` for this, you could do

```
symbolTableBuilder.definedClassNamesBuilder()
    .add(sourceClassName)
    .addAll(listInnerClassNames(javaClass));
```